linera-service/src/project.rs

```diff
@@ -251,4 +252,84 @@ impl Project {
             .expect("there was an error finding the version in a TOML file included at compile-time - this should never happen.");
         sdk_version.to_string()
     }
+
+    pub fn build(&self) -> Result<(PathBuf, PathBuf), anyhow::Error> {
+        let toml_path = self.root.join("Cargo.toml");
+        let linera_toml = LineraToml::new(toml_path)?;
+        let service_name = linera_toml.service_bin_name()?;
+        let contract_name = linera_toml.contract_bin_name()?;
+        let cargo_build = Command::new("cargo")
+            .arg("build")
+            .arg("--release")
+            .args(["--target", "wasm32-unknown-unknown"])
+            .current_dir(&self.root)
+            .spawn()?
+            .wait()?;
+        if !cargo_build.success() {
+            bail!("build failed")
+        }
+        let build_path = self.root.join("target/wasm32-unknown-unknown/release");
+        Ok((
+            build_path.join(contract_name).with_extension("wasm"),
+            build_path.join(service_name).with_extension("wasm"),
+        ))
+    }
+}
+
+struct LineraToml {
+    contents: toml::Value,
```

Comment:
Why define a wrapper around a dynamic value `toml::Value` instead of a usual Rust struct with meaningful fields and tagged with `#[derive(serde::Deserialize)]`?

Reply:
:facepalm: Good catch! Thanks :)